test/CMakeLists.txt

```diff
@@ -170,6 +165,25 @@ macro(onedpl_add_test test_source_file switch_off_checked_iterators)
     add_dependencies(build-onedpl-tests ${_test_name})
 endmacro()
 
+macro(onedpl_add_test test_source_file switch_off_checked_iterators)
+    set(_build_type_in_lower "unknown")
+    if (CMAKE_BUILD_TYPE)
+        string(TOLOWER "${CMAKE_BUILD_TYPE}" _build_type_in_lower)
+    endif()
+
+    get_filename_component(_test_name ${test_source_file} NAME)
+    string(REPLACE "\.cpp" "" _test_name ${_test_name})
+
+    onedpl_construct_exec(${test_source_file} ${_test_name} ${switch_off_checked_iterators} "")
+
+    ##if (_test_name STREQUAL "reduce.pass")
+    set(coal_tests "reduce.pass" "transform_reduce.pass" "count.pass")
+
+    if (_test_name IN_LIST coal_tests)
+        onedpl_construct_exec(${test_source_file} ${_test_name}.coal ${switch_off_checked_iterators} "-D_ONEDPL_DETECT_SPIRV_COMPILATION=0")
+    endif()
```

Comment:
Do we want to explicitly turn off (`-D_ONEDPL_DETECT_SPIRV_COMPILATION=1`) coalesced loads for the normal test path (or adjust the code specify each option for `coal_tests`, but not specify it across all tests)?  

If I understand correctly what is currently written, if we are running the tests on a non-SPIRV target, we will end up with 2 of the same test (named differently).  My initial thought was to have both versions always run regardless of the target.  
I understand that we may not care so much about the non-coalesced version on a non-SPIRV target, but the result as written may give a false impression of what the result of the tests are.

Reply:
I agree. Also, should we extend this testing approach to `sycl_iterator_reduce.pass` to catch errors during PR CI?

Reply:
I agree that that approach would be better. I have adjusted the CMake to build both versions regardless of target. I have also added `sycl_iterator_reduce.pass` to the list, let me know if the CI time increase is too large and ill remove it

Reply:
Thanks, @AidanBeltonS. Could you please rebase this PR to include #1319 which fixes the CI? We can then compare the CI times.

Reply:
Done

Reply:
Comparing the latest successful PR #1319 with this change, we see an accumulated increase in test time from about 3h to 3h 10m. The waiting times for developers won't change much with the slowest test (host, OpenMP backend, macOS) taking only 16 seconds more. The most affected are the DPCPP backend tests with 30-40 seconds increases with host offloading and 65 seconds increase on FPGA emulation. CI runtimes fluctuate a bit, so this is only an approximation. I think this increase is acceptable.